docs/tutorials/sinks/2_http_sink.md

```diff
@@ -0,0 +1,504 @@
+    type Request = BasicRequest;
+    type Error = std::io::Error;
+```
+
+## Metadata
+
+`Metadata` is the type for any information to be passed while building the request
+that is additional to the actual event being used. In this case we just need the
+`EventFinalizers`.
+
+## Events
+
+Events contains the type that is the event type passed to the `Encoder`.
+
+## Encoder
+
+The `Encoder` is the type that is used to encode the event to create the final
+payload. We are using the `BasicEncoder` described earlier.
+
+## Payload
+
+The `Payload` is the type of the final data that is encoded.
+
+## Request
+
+`Request` is the type that is sent to the final service. This is the `BasicRequest`
+we described earlier.
+
+## Error
+
+`Error` is the type of any errors that are creating while encoding the event.
+
+
+The following functions for the `RequestBuilder` trait need implementing:
+
+## compression
+
+```rust
+    fn compression(&self) -> Compression {
+        Compression::None
+    }
+```
+
+The payload for the built request can be compressed. Here we return `Compression::None`
+to indicate that we will not be compressing. `Compression::Gzip` and `Compression::Zlib`
+are available choices.
+
+
+## encoder
+
+```rust
+    fn encoder(&self) -> &Self::Encoder {
+        &self.encoder
+    }
+```
+
+We return the encoder to use. This is the `BasicEncoder` defined earlier.
+
+
+## split_input
+
+```rust
+    fn split_input(
+        &self,
+        mut input: Event,
+    ) -> (Self::Metadata, RequestMetadataBuilder, Self::Events) {
+        let finalizers = input.take_finalizers();
+        let metadata_builder = RequestMetadataBuilder::from_events(&input);
+        (finalizers, metadata_builder, input)
+    }
+```
+
+`split_input` takes the input and extracts the metadata from the events. In this
+case we are returning the `input` parameter unprocessed. 
+
+This may not always be the case. For example, the `amqp` sink will initially
+process  the event to extract fields to be used to calculate the exchange
+to send the message  to. This is sent  with the event to `split_input`.
+`split_input` splits that out into the event for  encoding and the metadata
+containing the exchange which will be used when sending  the event to `amqp`.
+
+
+## build_request
+
+```rust
+    fn build_request(
+        &self,
+        metadata: Self::Metadata,
+        request_metadata: RequestMetadata,
+        payload: EncodeResult<Self::Payload>,
+    ) -> Self::Request {
+        BasicRequest {
+            finalizers: metadata,
+            payload: payload.into_payload(),
+            metadata: request_metadata,
+        }
+    }
+```
+
+`build_request` is used to build the final request that will contain the encoded
+payload and the metadata. The `BasicRequest` object we return here is passed to 
+our `Tower` service where the data is actually sent.
+
+# Service
+
+We need to create a service that is responsible for actually sending our final
+encoded data.
+
```

Comment:
Worth mentioning that this is leveraging the aforementioned `Tower` service infrastructure, to tie it together.